src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -229,10 +236,13 @@ protected NodeContext GetNode(string msbuildLocation, string commandLineArgs, in
                     }
                 }
 #endif
-
-                // Create the node process
-                int msbuildProcessId = LaunchNode(msbuildLocation, commandLineArgs);
-                _processesToIgnore.Add(GetProcessesToIgnoreKey(hostHandshake, clientHandshake, msbuildProcessId));
+                int msbuildProcessId = -1;
+                lock (createNodeLock)
```

Comment:
I think you can remove the lock, everything in `LaunchNode` seems stack confined.